sympy/physics/optics/medium.py

```diff
@@ -152,6 +131,62 @@ def refractive_index(self):
         """
         return (c/self.speed)
 
+    def __lt__(self, other):
+        """
+        Compares based on refractive index of the medium.
+        """
+        return self.refractive_index < other.refractive_index
+
+    def __gt__(self, other):
+        return not self < other
+
+    def __ne__(self, other):
+        return not self == other
```

Comment:
I would just get rid of these. The `__ne__` method shouldn't be needed because Basic already defines it. The `__lt__` and `__gt__` methods are misguided and incorrect e.g. `__gt__` should be the negation of `self <= other` but that needs `__le__` which isn't defined. If someone wants to compare two `Medium` by their refractive index then they should just do that explicitly.

Reply:
Yes, that makes sense. I'll remove those methods.